.github/workflows/run-simulators.yml

```diff
@@ -0,0 +1,201 @@
+
+                if [[ "$system_status" == "ok" && "$instance_status" == "ok" ]]; then
+                    echo "Both SystemStatus and InstanceStatus are 'ok'"
+                    exit 0
+                fi
+
+                CURRENT_TIME=$(date +%s)
+                if [[ "$CURRENT_TIME" -ge "$END_TIME" ]]; then
+                    echo "Timeout: Both SystemStatus and InstanceStatus have not reached 'ok' state within $TIMEOUT seconds."
+                    exit 1
+                fi
+
+                sleep 10  # Check status every 10 seconds
+            done
+
+    check_simulator_version_updates:
+      name: check_simulator_version_updates
+      runs-on: ubuntu-latest
+      needs: start_ec2_instance
+      steps: 
+        - name: Check for Simulator Version Updates
+          env:
+            PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+            HOSTNAME: ${{ secrets.SSH_HOST }}
+            USER_NAME: ${{ secrets.SSH_USERNAME }}
+            GH_ACCESS_TOKEN: ${{ secrets.GH_ACCESS_TOKEN }}
+          run: |
+            echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+            ssh -o StrictHostKeyChecking=no -i private_key ${USER_NAME}@${HOSTNAME} '
+              cd /home/ubuntu/actions/ &&
+              rm -rf Scenic &&
+              git clone --branch $(basename "${{ github.ref }}") --single-branch https://$GH_ACCESS_TOKEN@github.com/BerkeleyLearnVerify/Scenic.git &&
+              cd Scenic &&
+              python3 -m venv venv &&
+              source venv/bin/activate &&
+              python3 -m pip install -e .[test-full] &&
+              python3 .github/check_latest_simulators.py
+            '
+
+    check_nvidia_smi:
+      name: check_nvidia_smi
+      runs-on: ubuntu-latest
+      needs: start_ec2_instance
+      continue-on-error: true
+      steps:
+        - name: Check NVIDIA SMI
+          env:
+            PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+            HOSTNAME: ${{ secrets.SSH_HOST}}
+            USER_NAME: ${{ secrets.SSH_USERNAME}}
+          run: |
+            echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+            ssh -o StrictHostKeyChecking=no -i private_key ${USER_NAME}@${HOSTNAME} '
+              output=$(nvidia-smi)
+              echo "$output"
+              if [ -z "$output" ]; then
+                echo "NVIDIA Driver is not set"
+                exit 1
+              fi
+            '  
+        - name: NVIDIA Driver is not set
+          if: ${{ failure() }}
+          run: |
+            echo "NVIDIA SMI is not working, please run the steps here on the instance:" 
+            echo "https://scenic-lang.atlassian.net/wiki/spaces/KAN/pages/2785287/Setting+Up+AWS+VM?parentProduct=JSW&initialAllowedFeatures=byline-contributors.byline-extensions.page-comments.delete.page-reactions.inline-comments.non-licensed-share&themeState=dark%253Adark%2520light%253Alight%2520spacing%253Aspacing%2520colorMode%253Alight&locale=en-US#Install-NVIDIA-Drivers"
```

Comment:
**Note:**  Adding this as a pre-req to run CARLA and webots tests. At times (unsure why), the NVIDIA driver seems to disconnect from the EC2 instance. Luckily, we have steps in place to re-install the necessary drivers to ensure that these simulator tests work. I am linking a Confluence Doc, that gives step by step instructions on what to run on the instance to re-install the drivers as needed!